test/unit/test_diagram.py

```diff
@@ -18,10 +18,34 @@
 # along with Autosubmit.  If not, see <http://www.gnu.org/licenses/>.
 
 """ Test file for autosubmit/monitor/diagram.py """
+
 import datetime
 
+from autosubmit.job.job import Job
+from autosubmit.monitor import diagram
 from autosubmit.monitor.diagram import JobData, JobAggData
 
+def test_seq():
+    """ function to test the Class JobData inside autosubmit/monitor/diagram.py """
+    for value in diagram._seq(100, 2, 10):
+        assert value is not None
+        assert isinstance(value, int)
+
+def test_populate_statistics():
+    """ function to test the Class JobData inside autosubmit/monitor/diagram.py """
+    jobs_data = [
+        Job('test', "a29z", "COMPLETED", 200),
+        Job('test', "a28z", "COMPLETED", 200),
+        Job('test', "a27z", "COMPLETED", 200),
+        Job('test', "a26z", "FAILED", 10)
+    ]
+
+    date_ini = datetime.datetime.now()
+    date_fin = date_ini + datetime.timedelta(0.10)
+    queue_time_fixes = ['test', 5]
+
+    statistics = diagram.populate_statistics(jobs_data,date_ini,date_fin, queue_time_fixes)
+    assert statistics is not None
```

Comment:
I think we could test a bit more. Like whether the statistics were populated (i.e. calculated) correctly.